### ISSUE TITLE
the "plugins" field in presets need to be an object

### ISSUE THREAD

**Issue by StarHeartHunt** [un-ts/eslint-plugin-import-x#79]
When i am migrating from eslintrc to flat config, i got the following error

```bash
> eslint .


Oops! Something went wrong! :(

ESLint: 9.2.0


A config object has a "plugins" key defined as an array of strings.

Flat config requires "plugins" to be an object in this form:

    {
        plugins: {
            import-x: pluginObject
        }
    }

Please see the following page for information on how to convert your config object into the correct format:
https://eslint.org/docs/latest/use/configure/migration-guide#importing-plugins-and-custom-parsers

If you're using a shareable config that you cannot rewrite in flat config format, then use the compatibility utility:
https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config

 ELIFECYCLE  Command failed with exit code 2.
```

After some research, i found the "plugins" key is defined in recommended preset:

https://github.com/un-ts/eslint-plugin-import-x/blob/8b53954f94f528b20e810f26cc562442761cf85d/src/config/recommended.ts#L7

Is it supposed to be a [virtual plugin](https://eslint.org/docs/latest/use/configure/plugins#configure-a-virtual-plugin) or a module? I'm glad to help improving it.

<details>
  <summary>Full code example</summary>

```js
import pluginImportX from "eslint-plugin-import-x";

export const imports = [
  pluginImportX.configs.recommended,
  pluginImportX.configs.typescript,
  {
    plugins: {
      import: pluginImportX,
    },
    rules: {
      "import-x/no-named-as-default": "off",
      "import-x/no-unresolved": ["error", { ignore: ["^virtual:"] }],
      "import-x/order": [
        "error",
        {
          groups: [
            "builtin",
            "external",
            "internal",
            "parent",
            "sibling",
            "index",
          ],
          pathGroups: [
            { pattern: "vue", group: "builtin", position: "before" },
          ],
          pathGroupsExcludedImportTypes: [],
          "newlines-between": "always",
          alphabetize: {
            order: "asc",
          },
        },
      ],
    },
  },
];

export default [
  ...imports
];
```
</details>

**Comment by JounQin** [comment]
The `configs` does not support flat config yet, see also #29

But you can config it manually by not using the `configs`

```js
import pluginImportX from "eslint-plugin-import-x";

export default [
  {
    plugins: {
      import: pluginImportX
    },
    rules: {
      "import/order": ["error"] // make sure to use same prefix as you defined above in `plugins`
    }
  }
]
```
